app/forms/placements/partnership_form.rb

```diff
@@ -0,0 +1,58 @@
+class Placements::PartnershipForm < ApplicationForm
+  attr_accessor :school_id, :provider_id, :javascript_disabled, :form_input
+
+  validate :id_presence
+  validate :validate_provider
+  validate :validate_school
+  validate :partnership_already_exists?
+
+  def provider
+    @provider ||= Placements::Provider.find_by(id: provider_id)
+  end
+
+  def school
+    @school ||= Placements::School.find_by(id: school_id)
+  end
+
+  def as_form_params
+    { "partnership" => { school_id: school.id, provider_id: provider.id } }
+  end
+
+  def persist
+    ActiveRecord::Base.transaction do
+      Placements::Partnership.create!(school:, provider:)
```

Comment:
Not sure why this needs to be in a base transaction. Usually, I use this when I want to revert everything if multiple saves need to happen.

Reply:
Resolved in [de46003](https://github.com/DFE-Digital/itt-mentor-services/pull/449/commits/de4600343e40b57cd20ba586b6321aec3fa9a686)